iDiff/utils/image_utils.go

```diff
@@ -70,6 +41,33 @@ func ImageToDir(img string, eng bool) (string, string, error) {
 	return TarToJSON(tarName)
 }
 
+// ImageToTar writes an image to a .tar file
+func ImageToTar(cli client.APIClient, image, tarName string) (string, error) {
+	imgBytes, err := cli.ImageSave(context.Background(), []string{image})
+	if err != nil {
+		return "", err
+	}
+	defer imgBytes.Close()
+	newpath := tarName + ".tar"
+	return newpath, copyToFile(newpath, imgBytes)
+}
+
+func CheckImageID(image string) bool {
+	pattern := regexp.MustCompile("[a-z|0-9]{12}")
+	if exp := pattern.FindString(image); exp != image {
+		return false
+	}
+	return true
+}
+
+func CheckImageURL(image string) bool {
+	pattern := regexp.MustCompile("^.+/.+(:.+){0,1}$")
+	if exp := pattern.FindString(image); exp != image || CheckTar(image) {
+		return false
+	}
+	return true
+}
+
 type Event struct {
```

Comment:
does this also make more sense in the docker_utils file since it is really only involved in Docker image pulls?